File: abc.com/client/src/app/roles/role.ts
import { Class } from '@angular/core';



export class Role {

  public  id: string;
  public  name: string;
  public lastUpadated: string;
  public  permissions: Array<string>;


  constructor() {
  }


  public setName ( name: string ) {
    this.name = name;
  }

  public getName(): string {
    return this.name;
  }

  public setLastUpadated ( lastUpadated: string ) {
    this.lastUpadated = lastUpadated;
  }

  public getLastUpadated (): string {
    return this.lastUpadated;
  }

  public setId ( id: string ) {
    this.id = id;
  }
  public getId (): string {
    return this.id ;
  }

  public setPermissions ( permissions: Array<string> ) {
    this.permissions = permissions;
  }
  public getPermissions (): Array<string> {
    return this.permissions ;
  }

  public  addPermission(permission:string) : void {
    if(this.permissions.find(item=> item === permission)) {
      return;
    }
    this.permissions.push(permission);
  }

  public  deletePermission(permission: string): void {
    let item = this.permissions.find(item => item === permission);
    const index: number = this.permissions.indexOf(item);
    if (index !== -1) {
      this.permissions.splice(index, 1);
    }
  }

}

File: abc.com/src/main/java/application/auth/roles/Role.java
package application.auth.roles;

import org.springframework.data.annotation.Id;

import application.common.BasicInfo;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by gipai on 9/28/2017.
 */
public class Role extends BasicInfo {

    @Id
    private String id;
    private String name;
    private List<PermissionType> permissions = new ArrayList<PermissionType>();

    public Role(){

    }

    public Role(boolean isSuperUser){
        if(isSuperUser){
            this.permissions.add(PermissionType.CREATE_USER);
            this.permissions.add(PermissionType.DELETE_USER);
            this.permissions.add(PermissionType.READ_USER);
            this.permissions.add(PermissionType.UPDATE_USER);

            this.permissions.add(PermissionType.CREATE_ROLE);
            this.permissions.add(PermissionType.DELETE_ROLE);
            this.permissions.add(PermissionType.READ_ROLE);
            this.permissions.add(PermissionType.UPDATE_ROLE);
        }
    }

    public Role(String name,List<PermissionType> permissions){
        this.name = name;
        this.permissions = permissions;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public List<PermissionType> getPermissions() {
        return permissions;
    }

    public void setPermissions(List<PermissionType> permissions) {
        this.permissions = permissions;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean addPermission(PermissionType permission) {
        int length = permissions.size();
        for(int i=0; i < length; i++){
            if(permissions.get(i) == permission){
                //return as already exists
                return false;
            }
        }
        permissions.add(permission);
        return true;
    }

    public boolean deletePermission(PermissionType permission) {
        int length = permissions.size();
        for(int i=0; i < length; i++){
            if(permissions.get(i) == permission){
                permissions.remove(i);
                return true;
            }
        }
        return false;
    }



}
